src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+
+            if (this->actor.bgCheckFlags & 1) {
+                this->actor.speedXZ = 0.0f;
+                this->actor.velocity.y = 3.5f;
+            }
+        }
+    }
+
+    if (this->timer4 != 0) {
+        Math_SmoothDownscaleMaxF(&this->unk_26C[9], 0.5f, 4000.0f);
+        iVar1 = 1;
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.x, this->unk_2B8, 1.0f, this->unk_2FC);
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.z, this->unk_2C0, 1.0f, this->unk_2FC);
+        pi = 0x8000 / M_PI;
+        Math_SmoothScaleMaxF(&this->unk_2FC, 3.0f, 1.0f, 0.3f);
+        fVar2 = this->unk_2B8 - this->actor.posRot.pos.x;
+        fVar3 = this->unk_2C0 - this->actor.posRot.pos.z;
+
+        if (fabsf(fVar2) < 10.0f) {
+            fVar2 = 0.0; // No f is required to match
+        }
+
+        if (fabsf(fVar3) < 10.0f) {
+            fVar3 = 0.0; // No f is required to match
+        }
+
+        if (fVar2 == 0.0f && fVar3 == 0.0f) {
+            this->timer4 = 0;
+            this->unk_29E = 7;
+        }
+
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, Math_atan2f(fVar2, fVar3) * pi, 3, this->unk_300, 0);
+        Math_SmoothScaleMaxF(&this->unk_300, 10000.0f, 1.0f, 1000.0f);
+    }
+
+    func_80AB5BF8(this, globalCtx, iVar1);
+}
+
+void func_80AB6A38(EnNiw* this, GlobalContext* globalCtx) {
+    Path* path;
+    Vec3s* pointPos;
+    f32 pathDiffX;
+    f32 pathDiffZ;
+    s16 pathNumber = this->path - 1;
+
+    if (this->path == 0) {
+        this->unk_2AC = this->unk_2B8 = this->actor.posRot.pos.x;
+        this->unk_2B0 = this->unk_2BC = this->actor.posRot.pos.y;
+        this->unk_2B4 = this->unk_2C0 = this->actor.posRot.pos.z;
+        this->timer5 = this->timer4 = this->unk_29E = 0;
+        this->unk_26C[7] = this->unk_26C[5] = this->unk_26C[6] = this->unk_26C[8] = this->actor.speedXZ =
+            this->unk_2FC = this->unk_300 = 0.0f;
+        this->actionFunc = EnNiw_ResetAction;
+    } else {
+        path = &globalCtx->setupPathList[pathNumber];
+        pointPos = SEGMENTED_TO_VIRTUAL(path->points);
+        pointPos += this->waypoint;
+        pathDiffX = pointPos->x - this->actor.posRot.pos.x;
+        pathDiffZ = pointPos->z - this->actor.posRot.pos.z;
+        this->unk_2E4 = Math_atan2f(pathDiffX, pathDiffZ) * 10430.378f;
+        func_80AB6100(this, globalCtx, 2);
+
+        if (fabsf(pathDiffX) < 30.0f && fabsf(pathDiffZ) < 30.0f) {
+            this->waypoint++;
+
+            if (this->waypoint >= this->unk_2EC) {
+                this->waypoint = 0;
+            }
+        }
+
+        func_80AB5BF8(this, globalCtx, 2);
+    }
+}
+
+void func_80AB6BF8(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->timer4 == 0) {
+        this->unk_2A6 = 2;
+        this->timer4 = 10;
+    }
+
+    VEC_SET(this->actor.shape.rot, Math_Rand_CenteredFloat(5000.0f), Math_Rand_CenteredFloat(5000.0f),
+            Math_Rand_CenteredFloat(5000.0f));
+
+    if (Actor_HasNoParent(&this->actor, globalCtx) != 0) {
```

Comment:
```suggestion
    if (Actor_HasNoParent(&this->actor, globalCtx)) {
```
`Actor_HasNoParent` returns a boolean so not having `!= 0` makes more sense